src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -432,9 +432,6 @@ public ReadOnlySequence<T> Slice(long start)
             if (start < 0)
                 ThrowHelper.ThrowStartOrEndArgumentValidationException(start);
 
-            if (start == 0)
```

Comment:
This seems like a legitimate short-circuit path. Why do we need to remove it?

Reply:
If we want to return the same result as others methods then this should be removed.  If we can return different results than this optimization can not changed.